mobile/src/main/java/ch/epfl/sdp/mobile/ui/features/profile/ProfileStateScreen.kt

```diff
@@ -0,0 +1,174 @@
+  Column(
+      modifier = modifier,
+      horizontalAlignment = Alignment.CenterHorizontally,
+  ) {
+    Spacer(Modifier.height(16.dp))
+    ProfilePicture(state)
+    Spacer(Modifier.height(16.dp))
+    Text(state.name, style = MaterialTheme.typography.h5)
+    Text(state.email, style = MaterialTheme.typography.subtitle2)
+    Spacer(Modifier.height(16.dp))
+    SettingsButton(onClick = state::onSettingsClick)
+    Spacer(Modifier.height(16.dp))
+  }
+}
+
+@Composable
+fun ProfilePicture(
+    state: ProfileState,
+    modifier: Modifier = Modifier,
+) {
+  Box(
+      modifier =
+          modifier.size(118.dp).background(state.backgroundColor.getColorForProfile(), CircleShape),
+      contentAlignment = Alignment.Center,
+  ) {
+    Text(state.emoji, style = MaterialTheme.typography.h3)
+    IconButton(
+        onClick = state::onEditClick,
+        modifier =
+            Modifier.align(Alignment.BottomEnd)
+                .background(MaterialTheme.colors.surface, CircleShape)
+                .border(2.dp, MaterialTheme.colors.primary, CircleShape)
+                .shadow(2.dp, CircleShape)
+                .size(40.dp),
+    ) { Icon(Icons.Default.Edit, "Edit profile icon") }
+  }
+}
+
+@Composable
+fun SettingsButton(onClick: () -> Unit, modifier: Modifier = Modifier) {
+  val strings = LocalLocalizedStrings.current
+  Button(
+      onClick = onClick,
+      shape = CircleShape,
+      contentPadding = PaddingValues(horizontal = 24.dp, vertical = 12.dp),
+      modifier = modifier) {
+    Icon(Icons.Default.Settings, null)
+    Spacer(modifier = Modifier.width(8.dp))
+    Text(strings.profileSettings)
+  }
+}
+
+@Composable
+fun infoComponent(text: String, num: Int, modifier: Modifier = Modifier) {
+  // Past games component
+  Column(
+      horizontalAlignment = Alignment.CenterHorizontally,
+      verticalArrangement = Arrangement.Center,
+      modifier = modifier) {
+    Text(text = text, modifier = modifier)
+    Text(
+        text = "$num",
+        style = TextStyle(textDecoration = TextDecoration.Underline),
+        modifier = modifier)
+  }
+}
+
+@Composable
+fun GamesInfo(numPastGames: Int, numPuzzles: Int, modifier: Modifier = Modifier) {
+  val strings = LocalLocalizedStrings.current
+  Row(
+      horizontalArrangement = Arrangement.Center,
+      verticalAlignment = Alignment.CenterVertically,
+      modifier = modifier) {
+    infoComponent(
+        text = strings.profilePastGames,
+        num = numPastGames,
+        modifier = modifier.padding(horizontal = 32.dp))
+    infoComponent(
+        text = strings.profilePuzzle,
+        num = numPuzzles,
+        modifier = modifier.padding(horizontal = 32.dp))
+  }
+}
+
+@OptIn(ExperimentalMaterialApi::class)
+@Composable
+fun Match(
+    title: String,
+    subtitle: String,
+    icon: ImageVector? = null,
+    modifier: Modifier = Modifier,
+) {
+  ListItem(
+      modifier = modifier,
+      icon = { Icon(Icons.Default.CheckBox, "Match icon") },

```

Comment:
Icons should be coming from extensions of the `PawniesIcons` object.

Reply:
👍